corehq/form_processor/tasks.py

```diff
@@ -9,6 +9,10 @@
 
 @no_result_task(queue=SUBMISSION_REPROCESS_CELERY_QUEUE, acks_late=True)
 def reprocess_submission(submssion_stub_id):
-    stub = UnfinishedSubmissionStub.objects.get(id=submssion_stub_id)
```

Comment:
probably worth wrapping this task in a `CriticalSection` to prevent race conditions